harness/tests/leader_election.rs

```diff
@@ -1,120 +1,68 @@
+        let arbitrary_follower_id = raft_list[0];
+
+        let follower_raft = rafts.get(&arbitrary_follower_id).unwrap().clone();
+        let leader_raft = rafts.get_mut(&leader_id).unwrap();
+
+        leader_raft.leave().await;
+        rafts.remove(&leader_id);
+        current_node_cnt -= 1;
+
+        let candidate_hash =
+            collect_state_matching_rafts_until_leader_emerge(&rafts, StateRole::Candidate).await;
+        let current_term = follower_raft
+            .raft_node
+            .storage()
+            .await
+            .hard_state()
+            .unwrap()
+            .get_term();
+
+        assert!(!raft_list.contains(&leader_id));
+
+        // Term must be increased after leader election
+        let new_leader_id = rafts.get(&raft_list[0]).unwrap().get_leader_id().await;
+        assert!(candidate_hash.get(&new_leader_id).is_some());
+
+        assert!(current_term > ex_term);
+        ex_term = current_term;
+        assert_eq!(current_node_cnt, follower_raft.get_cluster_size().await);
+    }
 
-    raft_2.quit().await;
-    let raft_3 = rafts.get_mut(&3).unwrap();
-    raft_3.quit().await;
+    for raft_id in raft_list {
+        rafts.get_mut(&raft_id).unwrap().quit().await;
+    }
 }
 
-// TODO: Fix this test.
 #[tokio::test]
-#[ignore]
 pub async fn test_leader_election_in_five_node_example() {
-    kill_previous_raft_processes();
-
-    let (tx_raft, rx_raft) = mpsc::channel::<(u64, Raft)>();
-    let peers = load_peers(FIVE_NODE_EXAMPLE).await.unwrap();
-    let _raft_tasks = tokio::spawn(build_raft_cluster(tx_raft, peers.clone()));
-
-    sleep(Duration::from_secs(1)).await;
-
-    let mut rafts = wait_until_rafts_ready(None, rx_raft, 5).await;
-
-    let raft_1 = rafts.get_mut(&1).unwrap();
-
-    wait_for_until_cluster_size_increase(raft_1.clone(), 5).await;
-
-    sleep(Duration::from_secs(1)).await;
-
-    raft_1.leave().await;
-
-    let raft_2 = rafts.get_mut(&2).unwrap();
-
-    wait_for_until_cluster_size_decrease(raft_2.clone(), 4).await;
-
-    sleep(Duration::from_secs(2)).await;
-
-    let leader_id = raft_2.get_leader_id().await;
-
-    assert!(
-        [2, 3, 4, 5].contains(&leader_id),
-        "Actual leader_id: {}",
-        leader_id
-    );
-
-    let leader_raft = rafts.get_mut(&leader_id).unwrap();
-    leader_raft.leave().await;
-
-    let mut remaining_nodes = vec![2, 3, 4, 5];
-    if let Some(pos) = remaining_nodes.iter().position(|&x| x == leader_id) {
-        remaining_nodes.remove(pos);
-    }
-
-    let raft_k = rafts.get_mut(&remaining_nodes[0]).unwrap();
-
-    wait_for_until_cluster_size_decrease(raft_k.clone(), 3).await;
-    sleep(Duration::from_secs(2)).await;
-
-    let leader_id = raft_k.get_leader_id().await;
-
-    assert!(leader_id != 0);
-    assert_eq!(raft_k.get_cluster_size().await, 3);
-
-    sleep(Duration::from_secs(2)).await;
+    run_leader_election_test(FIVE_NODE_EXAMPLE, vec![1, 2, 3, 4, 5]).await;
```

Comment:
Since the list of `node_id`s already exists in the example file, passing a separate list as an argument would be redundant.

Reply:
Thanks! I'll fix it